source/slang/slang-check-decl.cpp

```diff
@@ -1767,11 +1768,26 @@ namespace Slang
     {
         if (auto initExpr = varDecl->initExpr)
         {
+            // Disable the short-circuiting for static const variable init expression
+            if (varDecl->hasModifier<HLSLStaticModifier>() &&
+                varDecl->hasModifier<ConstModifier>())
+            {
+                disableShortCircuitLogicalExpr();
+            }
+
             // If the variable has an explicit initial-value expression,
             // then we simply need to check that expression and coerce
             // it to the type of the variable.
             //
             initExpr = CheckTerm(initExpr);
+
```

Comment:
This logic is problematic because we could be comming into this function with shortcircuiting disabled.
Instead of manually enable and disable here, why can't we create a new Context with `m_shouldShortCircuit` set to false and do `dispatchExpr` with that context?

Reply:
There is nothing preventing us from using subContext and dispatchExpr actually. I'm doing the manually enable and disable because it seems simpler without introducing more level of recursion.

> because we could be coming into this function with shortcircuiting disabled
Isn't it our goal here? We do want to have the shortcircuiting disabled here right?

I can change to subContext anyway. But I'm not quite understand the disadvantage of surrounding the `CheckTerm` by enable/disable pair. Can you please explain me more? Thanks.

Reply:
Because when you set a state such as m_shouldShortCircuit, you need to save its current value to a stack, and restore it after you've done with the sub-tree. The current value of `m_shortCircuit` is not necessarily `true` or `false` and we want to be robust. Instead of using an explicit stack data-structure, we just use the callstack to hold the previous state value. The subcontext mechanism accomplishes this goal.